### DISCUSSION TITLE
Datagrid and overlay button

### DISCUSSION THREAD

**Discussion by DomenicoFormoso** [Megabit/Blazorise#5515] [Q&A]
How to get grid row id when using overlay buttons and the row is not selected but only highlighted

```
<DataGrid TItem="Elemento"
	Data="@fogli"
    @bind-SelectedRow="@selectedFoglio"
    Responsive
    Virtualize
    Narrow="true"
    VirtualizeOptions="@(new() { DataGridHeight = "230px" })">
    <DataGridColumns>
    <DataGridCommandColumn />
    <DataGridColumn Field="@nameof(Elemento.Text)" Caption="Foglio" />
    </DataGridColumns>
     <RowOverlayTemplate>
            <Div>
		<Button Color="Color.Primary" Size="Size.Small" Clicked="ModificaFoglio">Modifica</Button>
		<Button Color="Color.Danger" Size="Size.Small">Elimina</Button>
                <Button Color="Color.Default" Size="Size.Small">Nuovo</Button>
            </Div>
        </RowOverlayTemplate>
</DataGrid>
```

selectedSheet is if I click on the button without selecting the row first.

**Comment by David-Moreira** [comment]
Currently the RowOverlayContext should allow you to access the row item. Is that what you are looking for?

So you should be able to set up delegates like so : 
`<Button Color="Color.Primary" Size="Size.Small" Clicked="() => ModificaFoglio(context.Item)">Modifica</Button>`

**Comment by DomenicoFormoso** [comment]
That's exactly what I wanted to do. Thank you very much, in my code:

` <Button Color="Color.Primary" Size="Size.Small" Clicked="(() => ModificaFoglio(context.Item.Id))">Modifica</Button>`
